sgkit/tests/test_distance.py

```diff
@@ -3,17 +3,30 @@
 import dask.array as da
 import numpy as np
 import pytest
+from numba import cuda
 from scipy.spatial.distance import (  # type: ignore
     correlation,
     euclidean,
     pdist,
     squareform,
 )
 
-from sgkit.distance.api import MetricTypes, pairwise_distance
+from sgkit.distance.api import MetricTypes, TargetTypes, pairwise_distance
 from sgkit.typing import ArrayLike
 
 
+def detect_cuda_driver() -> bool:
+    try:
+        return bool(len(cuda.list_devices()))
+    except cuda.CudaSupportError:
+        return False
+
+
+def skip_gpu_tests_if_no_gpu(target: TargetTypes) -> None:
+    if target == "gpu" and not detect_cuda_driver():
+        pytest.skip("Cuda driver not found")
+
+
 def get_vectors(
```

Comment:
Can this not be more like this?

```
def get_vectors(
    array_type: str = "da",
    dtype: str = "i8",
    size= (100, 100),
    chunk=(20, 10),
):
    rs = da.random.RandomState(0)
    x = rs.randint(0, 3, size=size, chunks=chunk, dtype=dtype)
    return x if array_type == "da" else np.asarray(x)
```

It makes the tests harder to follow IMO when the dask version is returning ints in [0, 2] and the numpy arrays are floats in [0, 1].